docs/devices_api.yml

```diff
@@ -180,36 +183,47 @@ definitions:
     properties:
       id:
         type: string
-      image:
+        description: Deployment ID (device unique)
+      artifact:
         type: object
         properties:
-          uri:
-            type: string
-          checksum:
-            type: string
-          id:
-            type: string
-          expire:
-            type: string
-            format: date-time
-          yocto_id:
+          source:
+            type: object
+            properties:
+              uri:
+                type: string
+                format: url
+                description: URL to fetch the image from
+              expire:
+                type: string
+                format: date-time
+                description: URL expiration time
+          device_types_compatible:
+            type: array
+            description: Compatible device types
+            items:
+              type: string
+          artifact_name:
             type: string
         required:
-          - uri
-          - id
-          - yocto_id
+          - source
+          - device_types_compatible
+          - artifact_name
     required:
-      - image
       - id
+      - artifact
     example:
       application/json:
-        image:
-          uri: 'https://aws.my_update_bucket.com/yocto_image123'
-          checksum: cc436f982bc60a8255fe1926a450db5f195a19ad
-          id: f81d4fae-7dec-11d0-a765-00a0c91e6bf6
-          expire: 2016-03-11T13:03:17.063493443Z
-          yocto_id: core-image-full-cmdline-20160330201408
         id: w81s4fae-7dec-11d0-a765-00a0c91e6bf6
+        artifact:
+          artifact_name: core-image-full-cmdline-20160330201408
```

Comment:
As we agreed to use human-readable name, maybe we can change this to `artifact_name: mender-1.1`?